### PR TITLE
 Created type definitions for "node-tlv"

### PR DESCRIPTION
**Pull request by David-Tennant**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/node-tlv/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": true
```
Comment (peterblazejewicz) [inline types/node-tlv/tsconfig.json:18]:
```suggestion
        "forceConsistentCasingInFileNames": true
```

Comment (peterblazejewicz) [reply types/node-tlv/tsconfig.json:18]: there is no real reason to use it here for DT, it's CJS module:
https://github.com/coolbong/node-tlv/blob/master/lib/TLV.js#L1067
It has sense for consumer tsconfig though.

File: types/node-tlv/node-tlv-tests.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
+import TLV from 'node-tlv';
+import assert from 'assert';
```
Comment (peterblazejewicz) [inline types/node-tlv/node-tlv-tests.ts:2]:
```suggestion
import TLV = require('node-tlv');
import assert = require('assert');
```
as per previosu comment about `esModuleInterop`

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@David-Tennant thx for the contribution!

**Comment by David-Tennant** [comment]
> LGTM!
> @David-Tennant thx for the contribution!

Thanks, @peterblazejewicz for the review!

**Comment by David-Tennant** [comment]
Ready to merge

**Pull request merged**